src/library_pipefs.js

```diff
@@ -0,0 +1,208 @@
+mergeInto(LibraryManager.library, {
+  $PIPEFS__postset: '__ATINIT__.push(function() { PIPEFS.root = FS.mount(PIPEFS, {}, null); });',
+  $PIPEFS__deps: ['$FS'],
+  $PIPEFS: {
+    BUCKET_BUFFER_SIZE: 1024 * 8, // 8KiB Buffer
+    mount: function (mount) {
+      return FS.createNode(null, '/', {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */, 0);
```

Comment:
I'm not sure, that it creates _identical_ root nodes: when `null` is passed as `parent` to `createNode`, as far as I understand, it creates a node that is parent of itself. And since all nodes have sequential inode numbers, then this pseudo-root node should have different parent inode compared to the "real" root directory. Looks like the only way new node is registered in `createNode` is through `FS.hashAddNode(...)` and all lookups through `FS.nameTable` are performed with comparison by name **and** parent id. Though, it seems that any call to `FS.unmount` will remove nodes, that are not linked to `FS.root`. Meanwhile `FS.mount` has special treatment for pseudo-mounts.